tools/clang/unittests/SPIRV/CodeGenSpirvTest.cpp

```diff
@@ -0,0 +1,30 @@
+//===- unittests/SPIRV/CodeGenSPIRVTest.cpp ---- Run CodeGenSPIRV tests ---===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "LibTestFixture.h"
+
+#include "gmock/gmock.h"
+
+namespace {
+using clang::spirv::LibTest;
+
+using ::testing::ContainsRegex;
+
+// This test is purely for demonstrating how to use `LibTest`, and does not
+// test anything specific in DXC itself.
+TEST_F(LibTest, SourceCodeWithoutFilePath) {
+  const std::string command(R"(// RUN: %dxc -T ps_6_0 -E PSMain -Zi)");
```

Comment:
Why keep the `// RUN:` pattern?
Is that to read filechecks tests and feed them as-is? Or would this be designed to only be called in tests like that?

If so, shall the command be passed to `compileCodeAndGetSpirvAsm(` in its own argument, saving us form the parsing logic?

What about an interface like:
```
compileCodeAndGetSpirvAsm("<shader-code>", "-Tcs_6_0 -Emain -fcgl -fspv-target-env=vulkan1.2")
```

Reply:
Even with that you will still have to keep the logic to part the options, which is most of the parsing code. 

I did it this way because I simply brought back the old code, and updated it to not use effcee. By minimizing the changes, I lower the chances of introducing a bug.

To me this is 6 of one and half a dozen of another. If you really insist I can change it, but I don't think either way is better or worse.

Also, this infra is to only be used when the test cannot be tested using the command line. We have had two cases for this in the past 6 years. I do not expect there to be many others, so do not want to put too much work into this.

Reply:
For this PR, I just want to resurrect the code that was removed with as few changes as possible. If you want to redesign it you can do that in a separate PR, but I don't think it is worth it.

Reply:
Right, I see the logic, and I'm fine with it.